src/gtest/test_dynamicusage.cpp

```diff
@@ -44,8 +46,10 @@ TEST(RecursiveDynamicUsageTests, TestTransactionJoinSplit)
 
     auto wtx = GetValidSproutReceive(sproutSk, 25000, true);
     // 2 vin + 1 vJoinSplit + 1 vShieldedOutput
-    // 160 + 1856 + 976
-    EXPECT_EQ(2992, RecursiveDynamicUsage(wtx));
+    // 160 + 1856 + 1200
+    EXPECT_EQ(0, wtx.GetSaplingSpendsCount());
+    EXPECT_EQ(1, wtx.GetSaplingOutputsCount());
+    EXPECT_EQ(3216, RecursiveDynamicUsage(wtx));
```

Comment:
```suggestion
    EXPECT_EQ(0, wtx.GetSaplingSpendsCount());
    EXPECT_EQ(1, wtx.GetSaplingOutputsCount());
    EXPECT_EQ(160 + 1856 + 1200, RecursiveDynamicUsage(wtx));
```

Reply:
+1 to this and the similar comments.

Even better if we have something like
```c++
CalcDynamicUsage(size_t numVin, size_t numVout, …)
```
or
```c++
2 * vinDynUsage + jsDynUsage + saplingOutputDynUsage
```

Reply:
I just went down an hour-long rabbithole trying to do this, and then remembered why we don't do this: we basically have to re-implement the entirety of `RecursiveDynamicUsage`, or we end up reusing it which makes the test pointless. The idea here is that we don't have any unexpected changes in memory usage, not that we calculate a specific value.

Reply:
I've just done @daira's suggested cleanups.